bot/src/bounded-contexts/analytics/infrastructure/ConsolidatedChartFactory.ts

```
@@ -0,0 +1,171 @@
+  /**
+   * Get legacy-compatible chart service
+   * This maintains backward compatibility with existing code
+   */
+  getLegacyChartService(): any {
+    const chartService = this.getChartService();
+    
+    // Return a proxy that adapts the new interface to the old one
+    return new Proxy({}, {
+      get: (_target, prop) => {
+        // Map old method names to new service
+        const methodMap: Record<string, Function> = {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)